tests/unit/test_bruteforce.cpp

```diff
@@ -1296,58 +1197,57 @@ TEST_F(BruteForceTest, brute_force_resolve_params) {
     array_free(rparams);
 }
 
-TEST_F(BruteForceTest, brute_get_distance) {
+TYPED_TEST(BruteForceTest, brute_get_distance) {
     size_t n = 4;
     size_t dim = 2;
     size_t numIndex = 3;
     VecSimIndex *index[numIndex];
     std::vector<double> distances;
 
-    float v1[] = {M_PI, M_PI};
-    float v2[] = {M_E, M_E};
-    float v3[] = {M_PI, M_E};
-    float v4[] = {M_SQRT2, -M_SQRT2};
+    TEST_DATA_T v1[] = {M_PI, M_PI};
+    TEST_DATA_T v2[] = {M_E, M_E};
+    TEST_DATA_T v3[] = {M_PI, M_E};
+    TEST_DATA_T v4[] = {M_SQRT2, -M_SQRT2};
 
-    VecSimParams params{
-        .algo = VecSimAlgo_BF,
-        .hnswParams = HNSWParams{.type = VecSimType_FLOAT32, .dim = dim, .initialCapacity = n}};
+    this->params.dim = dim;
+    this->params.initialCapacity = n;
 
     for (size_t i = 0; i < numIndex; i++) {
-        params.bfParams.metric = (VecSimMetric)i;
-        index[i] = VecSimIndex_New(&params);
+        this->params.metric = (VecSimMetric)i;
+        index[i] = CreateNewIndex(this->params);
         VecSimIndex_AddVector(index[i], v1, 1);
         VecSimIndex_AddVector(index[i], v2, 2);
         VecSimIndex_AddVector(index[i], v3, 3);
         VecSimIndex_AddVector(index[i], v4, 4);
         ASSERT_EQ(VecSimIndex_IndexSize(index[i]), 4);
     }
 
-    void *query = v1;
-    void *norm = v2;                                 // {e, e}
-    VecSim_Normalize(norm, dim, VecSimType_FLOAT32); // now {1/sqrt(2), 1/sqrt(2)}
-    ASSERT_FLOAT_EQ(((float *)norm)[0], 1.0f / sqrt(2.0f));
-    ASSERT_FLOAT_EQ(((float *)norm)[1], 1.0f / sqrt(2.0f));
+    TEST_DATA_T *query = v1;
+    TEST_DATA_T *norm = v2;                         // {e, e}
+    VecSim_Normalize(norm, dim, this->params.type); // now {1/sqrt(2), 1/sqrt(2)}
+    ASSERT_NEAR(norm[0], 1.0 / sqrt(2.0), 1e-5);
+    ASSERT_NEAR(norm[1], 1.0 / sqrt(2.0), 1e-5);
```

Comment:
please remind me why it was changed?

Reply:
ASSERT_FLOAT_EQ works well for both types but 
    ASSERT_DOUBLE_EQ fails for **floats**
so we can either stay with ASSERT_FLOAT_EQ or ASSERT_NEAR (which works well up to 1e-7

Reply:
Why not do it type dependent?

Reply:
done
check test_utils.h